docs/source/howto/codes.rst

```diff
@@ -45,8 +45,84 @@ Design guidelines
 Running external codes
 ======================
 
-`#3987`_
+To run an external code with AiiDA, you will need to use an appropriate :ref:`calculation plugin <topics:plugins>` that knows how to take transform the input nodes into the input files that the code expects, copy everything in the code's machine, run the calculation and retrieve the results.
+This plugin would have already been created, either by you (see section on :ref:`interfacing external codes <how-to:codes:plugin>`) or by other AiiDA plugin-developers (available packages are listed in the `plugin registry <https://aiidateam.github.io/aiida-registry/>`_), and has been properly installed in your environment.
+You can check which ones you have currently available by running the ``verdi plugin`` command:
 
+.. code-block:: bash
+
+    $ verdi plugin list aiida.calculations
+
+For the purpose of this example, we will use the ``arithmetic.add`` plugin, which is a plugin for bash to sum two integers that comes pre-installed with ``aiida-core``.
+You can access it through the use of the ``CalculationFactory``:
+
+.. code-block:: python
+
+    from aiida.plugins import CalculationFactory
+    calculation_class = CalculationFactory('arithmetic.add')
+
+We will obviously also need to provide the inputs for the code when running the calculation.
+You can also use ``verdi plugin`` to check this and other relevant information for specific plugins:
+
+.. code-block:: bash
+
+    $ verdi plugin list aiida.calculations arithmetic.add
+
+You will see that 3 inputs nodes are required: two containing the values to add up (``x``, ``y``) and one containing information about the specific code to execute (``code``).
+If you already have these nodes in your database, you can get them by :ref:`querying for them <how-to:data:finding-data>` or using ``orm.load_node(<PK>)``.
+Otherwise, you will need to create them as shown below (note that you `will` need to already have the ``localhost`` computer configured):
```

Comment:
We will need to provide a link where it is described how to do that or add a *dropdown* that shows this.

Reply:
You mean setting the localhost? Ok, yes, I agree. Would this be on the "Advanced Installation" section? I remember that when we were discussin the tutorial part of the assumptions was that they would have already had this set up, that's why I assumed it here.

Reply:
I think it hasn't been merged yet. So feel free to just ignore this for now. I'll try to add meaningful links to the relevant section after everything has been merged.

Reply:
Should we add a reminder of this...somewhere?

Reply:
I'll put it on the project board.